### ISSUE TITLE
[BUG]: SONIC_STP - does not match regular expression 

### ISSUE THREAD

**Issue by TafkaMax** [ansible-collections/dellemc.enterprise_sonic#457]
### Bug Description

When trying to use the sonic_stp module I am getting an error regarding the regular expression.  "does not match regular expression pattern"

### Product Name

Dell Enteprise Sonic

### Component or Module Name

httpapi

### DellEMC Enterprise SONiC Ansible Collection Version

main - commit: 6eafbcdeec5c675ee4cdcd10920c6ee51edc3348

### SONiC Software Version

4.4.0

### Configuration

```
CONFIG_FILE() = /home/tansper/ansible/ansible.cfg
DEFAULT_HOST_LIST(/home/tansper/ansible/ansible.cfg) = ['/home/tansper/ansible/hosts']
DEFAULT_LOOKUP_PLUGIN_PATH(/home/tansper/ansible/ansible.cfg) = ['/home/tansper/ansible/plugins/lookup']
DEFAULT_VAULT_PASSWORD_FILE(/home/tansper/ansible/ansible.cfg) = /home/tansper/ansible/scripts/vault_password.sh
EDITOR(env: EDITOR) = vim
HOST_KEY_CHECKING(/home/tansper/ansible/ansible.cfg) = False
PAGER(env: PAGER) = less
```

### Steps to Reproduce

1. Use sonic_stp module
2. Specify the config as such:
``` 
{
        "global": {
            "bpdu_filter": true,
            "bridge_priority": 4096,
            "enabled_protocol": "mst",
            "loop_guard": true
        },
        "mstp": {
            "mst_instances": [
                {
                    "bridge_priority": 4096,
                    "mst_id": 1,
                    "vlans": [
                        "20",
                        "90"
                    ]
                },
                {
                    "bridge_priority": 4096,
                    "mst_id": 2,
                    "vlans": [
                        "22",
                        "70",
                        "72",
                        "91"
                    ]
                },
                {
                    "bridge_priority": 4096,
                    "mst_id": 3,
                    "vlans": [
                        "21",
                        "23",
                        "69",
                        "71",
                        "78",
                        "92"
                    ]
                }
            ],
            "mst_name": "default-mst",
            "revision": 1
        }
    }

```

### Expected Behavior

MSTP is set accordingly

### Actual Behavior

The module fails with "does not match regular expression pattern"

### Logs

```
{
  "changed": false,
  "code": -32603,
  "msg": {
    "ietf-restconf:errors": {
      "error": [
        {
          "error-type": "protocol",
          "error-tag": "invalid-value",
          "error-message": "/device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"78\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"70\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"22\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"21\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"72\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"23\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"69\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"71\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"90\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"91\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"20\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\", /device/stp/mstp: /device/stp/mstp/mst-instances: /device/stp/mstp/mst-instances/mst-instance: /device/stp/mstp/mst-instances/mst-instance/config/vlan: schema \"\": \"92\" does not match regular expression pattern \"^(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])\\\\\\\\.\\\\\\\\.(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{1,2}|[1-9])$\""
        }
      ]
    },
    "code": 400,
    "request_data": {
      "path": "data/openconfig-spanning-tree:stp/mstp",
      "method": "patch"
    }
  }
}
```

**Comment by kerry-meyer** [comment]
Thank you very much for bringing this problem to our attention.

By using the example case you provided, we were able to find the bug that was causing the symptoms you have described.

The fix for this problem (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/460 ) was merged to the "main" branch of the Github repo for the Dell enterprise_sonic Ansible resource module collection today and is available for installation via the method described at https://github.com/ansible-collections/dellemc.enterprise_sonic/wiki/Installation-of-the-latest-pre-release-version .

I will wait for your confirmation of the fix before closing this Issue.
